src/clientdynamicinfo.cpp

```diff
@@ -37,19 +38,22 @@ ClientDynamicInfo ClientDynamicInfo::getCurrent()
 
     return {
         screen_size, real_gui_scaling, real_hud_scaling,
-        ClientDynamicInfo::calculateMaxFSSize(screen_size, gui_scaling),
+        ClientDynamicInfo::calculateMaxFSSize(screen_size, density, gui_scaling),
         touch_controls
     };
 }
 
-v2f32 ClientDynamicInfo::calculateMaxFSSize(v2u32 render_target_size, f32 gui_scaling)
+v2f32 ClientDynamicInfo::calculateMaxFSSize(v2u32 render_target_size, f32 density, f32 gui_scaling)
 {
-    f32 factor = (g_settings->getBool("touch_gui") ? 10 : 15) / gui_scaling;
-    f32 ratio = (f32)render_target_size.X / (f32)render_target_size.Y;
-    if (ratio < 1)
-        return { factor, factor / ratio };
-    else
-        return { factor * ratio, factor };
+	// must stay in sync with GUIFormSpecMenu::calculateImgsize
```

Comment:
You can remove this comment now.

Reply:
I prefer to keep these comments since they're still useful, both as reminders and as quick links.